src/Integration.Vsix/Analysis/UserSettingsProvider.cs

```diff
@@ -70,7 +69,15 @@ private void OnFileChanged(object sender, EventArgs e)
 
         #region IUserSettingsProvider implementation
 
-        public UserSettings UserSettings { get; private set; }
+        public UserSettings UserSettings
+        {
+            get
```

Comment:
I get this -> https://rules.sonarsource.com/csharp/RSPEC-2190/ for both the set and get (Add a way to break out of this property) and I don't fully understand considering this is not like, a loop or anything

Reply:
`if (UserSettings == null)...` in line 76 will call the `get` method from inside the `get` method -> recursion. Have you tried debugging it?

Reply:
No, for some reason on my machine it wont run the tests for this class (I click run and something compiles and then nothing happens) so I wanted to upload it to see if it runs there. And yes your right, In my head I had added another private variable to use for checking and assigning but clearly that did not happen